sdk/src/types/block/semantic.rs

```diff
@@ -342,6 +313,26 @@ impl<'a> SemanticValidationContext<'a> {
                     .checked_add(consumed_native_token.amount())
                     .ok_or(Error::ConsumedNativeTokensAmountOverflow)?;
             }
+
+            if let Some(unlocks) = self.unlocks {
+                if unlocks.len() != self.inputs.len() {
+                    return Ok(Some(TransactionFailureReason::InvalidInputUnlock));
```

Comment:
This error is not really correct if one is just missing

Reply:
There is no better error 🤷🏻‍♂️

Reply:
I don't like this limited API error things